oshmem/shmem/c/shmem_wait.c

```diff
@@ -89,11 +89,18 @@
 
 SHMEM_TYPE_WAIT(, volatile long, SHMEM_LONG, shmem)
 SHMEM_TYPE_WAIT(_short, volatile short, SHMEM_SHORT, shmem)
+SHMEM_TYPE_WAIT(_ushort, volatile unsigned short, SHMEM_USHORT, shmem)
 SHMEM_TYPE_WAIT(_int, volatile int, SHMEM_INT, shmem)
+SHMEM_TYPE_WAIT(_uint, volatile unsigned int, SHMEM_UINT, shmem)
 SHMEM_TYPE_WAIT(_long, volatile long, SHMEM_LONG, shmem)
+SHMEM_TYPE_WAIT(_ulong, volatile unsigned long, SHMEM_ULONG, shmem)
 SHMEM_TYPE_WAIT(_longlong, volatile long long, SHMEM_LLONG, shmem)
+SHMEM_TYPE_WAIT(_ulonglong, volatile unsigned long long, SHMEM_LLONG, shmem)
 SHMEM_TYPE_WAIT(_int32, int32_t, SHMEM_INT32_T, shmemx)
-SHMEM_TYPE_WAIT(_int64, int64_t, SHMEM_INT64_T, shmemx)
+SHMEM_TYPE_WAIT(_uint32, uint32_t, SHMEM_UINT32_T, shmemx)
+SHMEM_TYPE_WAIT(_uint64, uint64_t, SHMEM_UINT64_T, shmemx)
+SHMEM_TYPE_WAIT(_size, size_t, SHMEM_SIZE_T, shmem)
+SHMEM_TYPE_WAIT(_ptrdiff, ptrdiff_t, SHMEM_PTRDIFF_T, shmem)
```

Comment:
actually for unsigned types, shmem_wait is not defined, so not needed
https://github.com/open-mpi/ompi/blob/ef7ab9bf9699fa81ba27aed7a32305e9f3e313c4/oshmem/include/shmem.h.in#L1639-L1643

Reply:
Isn't it a problem by itself?

Reply:
it's not required by openshmem spec
http://www.openshmem.org/site/sites/default/site_files/OpenSHMEM-1.5.pdf